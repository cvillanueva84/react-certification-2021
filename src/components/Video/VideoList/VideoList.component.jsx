import React from 'react';
import { videoData } from '../../../youtube-videos-mock';
import { Container } from './VideoList.styled';
import VideoItem from '../VideoItem/VideoItem.component';

export default function VideoList() {
  const { items } = videoData;

  const avatar = items[0].snippet.thumbnails.default.url;

  return (
    <Container>
      {items.slice(1).map((item, id) => {
        const {
          snippet: {
            thumbnails: {
              medium: { url },
            },
          },
        } = item;

        const {
          snippet: { title },
        } = item;

        const {
          snippet: { description },
        } = item;

        return (
          <VideoItem
            key={id}
            url={url}
            title={title}
            description={description}
            avatar={avatar}
          />
        );
      })}
    </Container>
  );
}
