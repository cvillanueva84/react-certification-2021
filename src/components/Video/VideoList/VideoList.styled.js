import styled from 'styled-components';

export const Container = styled.section`
  margin-top: 100px;
  position: relative;
  z-index: 0;
  width: 800px;

  @media screen and (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-column-gap: 1.5rem;
  }

  @media screen and (min-width: 992px) {
    grid-template-columns: repeat(3, 1fr);
  }
`;
