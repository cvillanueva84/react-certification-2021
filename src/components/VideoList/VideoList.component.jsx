import React from 'react';
import { Link } from 'react-router-dom';
import {
  VideoCardDiv,
  VideoImageDiv,
  TitleSpan,
  DescriptionSpan,
} from './styledComponents';

const VideoList = ({ video }) => {
  const { snippet } = video;
  const { thumbnails, title, description } = snippet;
  const { id } = video;
  const { videoId } = id;
  return (
    <Link to={`/video/${videoId}`} style={{ textDecoration: 'none', color: '#000' }}>
      <VideoCardDiv>
        <VideoImageDiv
          style={{
            backgroundImage: `url(${thumbnails.medium.url})`,
            backgroundSize: 'cover',
            backgroundRepeat: 'no-repeat',
            backgroundPosition: 'center',
            borderTopRightRadius: '0.4rem',
            borderTopLeftRadius: '0.4rem',
          }}
        ></VideoImageDiv>
        <TitleSpan>{title}</TitleSpan>
        <DescriptionSpan>{description ? description : 'watch video...'}</DescriptionSpan>
      </VideoCardDiv>
    </Link>
  );
};

export default VideoList;
