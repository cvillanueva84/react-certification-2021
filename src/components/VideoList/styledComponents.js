import styled from 'styled-components';

export const VideoCardDiv = styled.div`
  display: flex;
  flex-direction: column;
  width: 30rem;
  height: 35rem;
  border-radius: 0.4rem;
  margin: 2rem;
  border: 1px #dad4d4 solid;
  transition: all 0.3s ease;
  &:hover {
      box-shadow: 10px 10px 24px 0px rgba(0, 0, 0, 0.75);
    }
    
    @media only screen and (max-width: 670px) {
        width: 80%;
    }
`;
VideoCardDiv.displayName = 'VideoCardDiv';

export const VideoImageDiv = styled.div`
  height: 140px;
`;
VideoImageDiv.displayName = 'VideoImageDiv';

export const TitleSpan = styled.span`
  font-size: 1.6rem;
  font-weight: bold;
  margin: 2rem;
`;
TitleSpan.displayName = 'TitleSpan';

export const DescriptionSpan = styled.span`
  font-size: 1.4rem;
  margin-bottom: 2rem;
  text-align: left;
  margin: 2rem;
  `;
DescriptionSpan.displayName = 'DescriptionSpan';

