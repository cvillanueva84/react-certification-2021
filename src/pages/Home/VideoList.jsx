import React from 'react';
import styled from 'styled-components';
import VideoItem from './VideoItem';

const VideoListContainer = styled.div`
    width: 100%;
    height: 100vh;
    margin: 5px auto;
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    
    & a {
        text-decoration: none;
    }
`;

const VideoList = ({videos}) => {

    if (videos.length < 0 ) return null;

    return (
        <VideoListContainer>
            {videos.items.map( (item) => (
                <VideoItem item={item} key={item.etag} />
            ))}
        </VideoListContainer>
    );

};

 export default VideoList;

 