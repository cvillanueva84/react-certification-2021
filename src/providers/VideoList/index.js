export { default, useAuth } from './Search.provider';
