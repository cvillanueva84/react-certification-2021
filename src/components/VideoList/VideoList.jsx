import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { DivTitle, Container, Fakegrid } from './VideoList.styled';
import Card from '../Card/Card';

const baseURL =
  'https://gist.githubusercontent.com/jparciga/1d4dd34fb06ba74237f8966e2e777ff5/raw/f3af25f1505deb67e2cc9ee625a633f24d8983ff/youtube-videos-mock.json';

const VideoList = () => {
  const [videos, setVideos] = useState([]);

  useEffect(() => {
    async function getVideos() {
      const {
        data: { items },
      } = await axios.get(baseURL);
      const aux = [...items];
      aux.shift();
      setVideos(aux);
      // console.log(data.items.map((el) => el.snippet.description));
    }
    getVideos();
  }, []);

  return (
    <>
      <DivTitle>
        <h2>Welcome to the Challenge!</h2>
      </DivTitle>
      <Container>
        <Fakegrid>
          {videos.map((video) => (
            <Card
              title={video.snippet.title}
              description={video.snippet.description}
              link={video.snippet.thumbnails.high.url}
              key={video.snippet.publishedAt}
            />
          ))}
        </Fakegrid>
      </Container>
    </>
  );
};

export default VideoList;
