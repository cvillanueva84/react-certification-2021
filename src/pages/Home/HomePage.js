import React, { useContext, Suspense, lazy } from 'react';
import { SectionContainer, VideoListContainer, override } from './HomePage.styled';
import { VideoListContext } from '../../providers/VideoList/VideoList.provider';
import GridLoader from 'react-spinners/ClipLoader';
const VideoList = lazy(() => import('../../components/VideoList/VideoListComponent'));

const HomePage = () => {
  const { videosState } = useContext(VideoListContext);
  const { posts = [] } = videosState;

  return (
    <SectionContainer>
      <div>
        <h2 style={{ marginLeft: '4rem', marginTop: '2rem' }}>Our videos:</h2>
      </div>
      <Suspense fallback={<GridLoader size={150} css={override} />}>
        <VideoListContainer>
          {posts.map((video) => (
            <VideoList key={video.id.videoId} video={video} />
          ))}
        </VideoListContainer>
      </Suspense>
    </SectionContainer>
  );
};

export default HomePage;
