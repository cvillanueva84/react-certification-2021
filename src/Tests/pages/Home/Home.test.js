import React from 'react'
import { mount} from 'enzyme'
import HomePage from '../../../pages/Home/Home.page';
import { VideoListContext } from '../../../providers/VideoList/VideoList.provider';
import { videoList as videos} from '../../mockData/videosList'

describe('Tests of the < HomePage/> component', () => {
    
    const wrapper = mount(
        <VideoListContext.Provider
            value={{
                videos
            }}
        >
            < HomePage/>
        </VideoListContext.Provider>)


    test('should show the < HomePage/> component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should contain the SectionContainer styled component', () => {
        expect(wrapper.find('SectionContainer').length).toEqual(1);
    })
    

    test('should contain the VideoListContainer styled component', () => {
        expect(wrapper.find('VideoListContainer').length).toEqual(1);

    })
})
