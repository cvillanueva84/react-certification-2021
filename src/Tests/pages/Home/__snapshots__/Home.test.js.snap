// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of the < HomePage/> component should show the < HomePage/> component correctly 1`] = `
<section
  className="sc-bdnxRM gmHjgD"
>
  <div>
    <h2
      style={
        Object {
          "marginLeft": "4rem",
          "marginTop": "2rem",
        }
      }
    >
      Our videos:
    </h2>
  </div>
  <Suspense
    fallback={
      <Loader
        color="#000000"
        css={
          Object {
            "map": undefined,
            "name": "1x3mwo8",
            "next": undefined,
            "styles": "
  display: block;
  margin: 20rem auto;
  border-color: #060b26;
",
            "toString": [Function],
          }
        }
        loading={true}
        size={150}
        speedMultiplier={1}
      />
    }
  >
    <div
      className="sc-gtsrHT evurdI"
    />
    <span
      className="css-17hjzrw"
    />
  </Suspense>
</section>
`;
